onebusaway-android/src/google/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -233,10 +234,14 @@ private void initMap(Bundle savedInstanceState) {
             initMapState(savedInstanceState);
         } else {
             Bundle args = getActivity().getIntent().getExtras();
-            // The rest of this code assumes a bundle exists,
-            // even if it's empty
+            // The rest of this code assumes a bundle exists, even if it's empty
             if (args == null) {
                 args = new Bundle();
+                Location l = getMapCenterAsLocation();
+                if (l == null || l.getLatitude() == 0.0 || l.getLongitude() == 0.0) {
+                    // We don't have a map view location yet - try getting last one from prefs
+                    PreferenceUtils.maybeRestoreMapViewToBundle(args);
```

Comment:
@cagryInside Could you please try debugging this and seeing if the code makes it into this IF branch on your device?  It's possible that this could be initializing differently on different devices, or versions of Google Play Services.  We may want to completely remove the IF here and always execute PreferenceUtils.maybeRestoreMapViewToBundle().